csharp/src/Google.Protobuf/UnknownFieldSet.cs

```diff
@@ -0,0 +1,340 @@
+			}
+			lastFieldNumber = number;
+			return lastField;
+		}
+
+		/// <summary>
+		/// Adds a field to the set. If a field with the same number already exists, it
+		/// is replaced.
+		/// </summary>
+		public UnknownFieldSet AddField(int number, UnknownField field)
+		{
+			if (number == 0)
+			{
+				throw new ArgumentOutOfRangeException("number", "Zero is not a valid field number.");
+			}
+			fields[number] = field;
+			return this;
+		}
+
+		/// <summary>
+		/// Resets to an empty set.
+		/// </summary>
+		public UnknownFieldSet Clear()
+		{
+			fields.Clear();
+			lastFieldNumber = 0;
+			lastField = null;
+			return this;
+		}
+
+		/// <summary>
+		/// Parse an entire message from <paramref name="input"/> and merge
+		/// its fields into this set.
+		/// </summary>
+		public void MergeFrom(CodedInputStream input)
+		{
+			uint tag;
+			tag = input.ReadTag();
+			while (tag != 0)
+			{
+				if (!MergeFieldFrom(input))
+				{
+					break;
+				}
+				tag = input.ReadTag();
+			}
+		}
+
+		/// <summary>
+		/// Parse a single field from <paramref name="input"/> and merge it
+		/// into this set.
+		/// </summary>
+		/// <param name="input">The coded input stream containing the field</param>
+		/// <returns>false if the tag is an "end group" tag, true otherwise</returns>
+		public bool MergeFieldFrom(CodedInputStream input)
+		{
+			uint tag = input.LastTag;
+			int number = WireFormat.GetTagFieldNumber(tag);
+			switch (WireFormat.GetTagWireType(tag))
+			{
+				case WireFormat.WireType.Varint:
+					{
+						ulong uint64 = input.ReadUInt64();
+						GetField(number).AddVarint(uint64);
+						return true;
+					}
+				case WireFormat.WireType.Fixed32:
+					{
+						uint uint32 = input.ReadFixed32();
+						GetField(number).AddFixed32(uint32);
+						return true;
+					}
+				case WireFormat.WireType.Fixed64:
+					{
+						ulong uint64 = input.ReadFixed64();
+						GetField(number).AddFixed64(uint64);
+						return true;
+					}
+				case WireFormat.WireType.LengthDelimited:
+					{
+						ByteString bytes = input.ReadBytes();
+						GetField(number).AddLengthDelimited(bytes);
+						return true;
+					}
+				default:
+					throw new InvalidOperationException("Wire Type is invalid.");
+			}
+		}
+
+		/// <summary>
+		/// Merges the fields from <paramref name="other"/> into this set.
+		/// If a field number exists in both sets, the values in <paramref name="other"/>
+		/// will be appended to the values in this set.
+		/// </summary>
+		public UnknownFieldSet MergeFrom(UnknownFieldSet other)
+		{
+			if (other != DefaultInstance)
+			{
+				foreach (KeyValuePair<int, UnknownField> entry in other.fields)
+				{
+					MergeField(entry.Key, entry.Value);
+				}
+			}
+			return this;
+		}
+
+		/// <summary>
+		/// Adds a field to the unknown field set. If a field with the same
+		/// number already exists, the two are merged.
+		/// </summary>
+		public UnknownFieldSet MergeField(int number, UnknownField field)
```

Comment:
This doesn't looks like it needs to be public.

Reply:
done